autox/environment/nncf/quantization_env.py

```diff
@@ -0,0 +1,586 @@
+            feature['kernel']           = 1.0
+            feature['param']            = np.prod(m.weight.size())     
+            feature['ifm_size']         = np.prod(m._input_shape[-1]) # feature nodes
+            feature['prev_action']      = 0.0 # placeholder      
+
+        else:
+            assert len(m._input_shape) in [2,4] , "new condition encountered, pls revise design"
+
+            if row['qid'] == 'MobileNetV2/adaptive_avg_pool2d_0':
+                qm = row['qmodule']
+                if len(m._input_shape) == 2:
+                    feature['cin']              = qm._input_shape[-1]
+                    feature['cout']             = qm._output_shape[-1]
+                    feature['ifm_size']         = qm._input_shape[-1]
+                elif len(m._input_shape) == 4:
+                    feature['cin']              = qm._input_shape[1]
+                    feature['cout']             = qm._output_shape[1]
+                    feature['ifm_size']         = np.prod(qm._input_shape[-2:]) # H*W
+            else:
+                if len(m._input_shape) == 2:
+                    feature['cin']              = m._input_shape[-1]
+                    feature['cout']             = m._output_shape[-1]
+                    feature['ifm_size']         = m._input_shape[-1]
+                elif len(m._input_shape) == 4:
+                    feature['cin']              = m._input_shape[1]
+                    feature['cout']             = m._output_shape[1]
+                    feature['ifm_size']         = np.prod(m._input_shape[-2:]) # H*W
+
+            feature['conv_dw']          = 0.0 
+            feature['stride']           = 0.0
+            feature['kernel']           = 0.0 # what should we set for non-learnable layer?
+            feature['param']            = 0.0     
+            feature['prev_action']      = 0.0 # placeholder 
+           
+        return pd.Series(feature)
+       
+    # def _create_quantization_pipeline(self, nncf_config):
+    #     _pretrained_model = deepcopy(self.pretrained_model)
+    #     compression_ctrl, compressed_model = create_compressed_model(_pretrained_model, nncf_config)
+    #     return compression_ctrl, compressed_model
+
+
+    def _run_quantization_pipeline(self, finetune):
+        self._adaptbn(self.config)
+        
+        if finetune:
+            # TODO: need to fix finetune loop
+            top1, top5 = self._finetune(self.qctrl, self.qmodel)
+            logger.info("[Q.Env] Post-Tuned: Top1@ {:.3f}, Top5@ {:.3f}".format(top1, top5))
+        else:
+            with torch.no_grad():
+                quantized_acc = self.validate_fn(self.val_loader, self.qmodel, self.criterion, self.config)
+                logger.info("[Q.Env] Post-Init: {:.3f}".format(quantized_acc))
+
+        return quantized_acc
+
+
+    def _adaptbn(self, config):
+        train_mode = self.qmodel.training
+        if not train_mode:
+            self.qmodel.train()
```

Comment:
it seems `run_batchnorm_adaptation` should call it inside, otherwise, it would work incorrectly @vanyalzr

Reply:
need advise on this too @vanyalzr . But when I was developing this method, i ablated BNAdap, i did observe changes in accuracy.

Reply:
Low priority. @vanyalzr should fix that.